joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/TypeNameProvider.scala

```diff
@@ -163,31 +163,75 @@ trait TypeNameProvider { this: AstCreator =>
     }
   }
 
+  private val FundamentalTypeKeywords = List(
+    "void",
+    "bool",
+    "true",
+    "false",
```

Comment:
this is... weird? If it is correct / intended, please add some comments explaining what's going on with these being designated as type keywords.

Reply:
That comes from the official C++ docs.

Reply:
what link? I don't see one for `FundamentalTypeKeywords`. The one below for `isFundamentalTypeKeywords` does mention `true` and `false` - but only as the possible values for `bool` not as types themselves.

Reply:
Ah you meant true/false explicitely. Right, thats a copy/paste error.